[io] properly stream TClonesArray with empty slots

ferdymercury opened the pull request:
# This Pull request:

## Changes or fixes:

Fixes https://its.cern.ch/jira/browse/ROOT-6788 by @olifre

This fixes the crash, but not sure if it is correct.

## Checklist:

- [x] tested changes locally
- [ ] updated the docs (if necessary)

ferdymercury committed:
[io] properly stream TClonesArray with empty slots

Fixes https://its.cern.ch/jira/browse/ROOT-6788

pcanal commented on io/io/src/TStreamerInfoActions.cxx:
```diff
@@ -2491,7 +2491,8 @@ namespace TStreamerInfoActions
          for(void *iter = start; iter != end; iter = (char*)iter + sizeof(void*) ) {
-            action(buf, *(void**)iter, config);
+            if (*(void**)iter)
+               action(buf, *(void**)iter, config);
```
The issue with this solution is that it change the index of the element when they are read back (what is written *_is_* different from what is read).

ferdymercury committed:
[io] do not change order of elements if null

attempt to apply pcanal's suggestion

Apply suggestion from @pcanal

Co-authored-by: Philippe Canal <pcanal@fnal.gov>

apply pcanals hints
